### PR TITLE
Bug 1125850 - Use lxml to render sitemap XML.

### PR DESCRIPTION
**Pull request by jezdez**
That also reduces the risk of memory leakage when building all sitemaps.

### CODE REVIEW

File: kuma/wiki/tasks.py [hunk 1]
```diff
@@ -265,13 +265,17 @@ class WikiSitemap(GenericSitemap):
...
+SITEMAP_ELEMENT = u'<sitemap><loc>%s</loc><lastmod>%s</lastmod></sitemap>'
+
+
 @task
 def build_sitemaps():
-    sitemap_element = '<sitemap><loc>%s</loc><lastmod>%s</lastmod></sitemap>'
-    sitemap_index = ('<sitemapindex xmlns="http://www.sitemaps.org/'
-                     'schemas/sitemap/0.9">')
+    sitemap_parts = [(u'<sitemapindex xmlns="http://www.sitemaps.org/'
+                      u'schemas/sitemap/0.9">')]
```
Comment (robhudson) [inline kuma/wiki/tasks.py]: looks like you don't need the parents now that you have the list brackets

File: kuma/wiki/tasks.py [hunk 2]
```diff
@@ -281,25 +285,29 @@ def build_sitemaps():
...
             if not os.path.exists(directory):
                 os.makedirs(directory)
+
             with open(os.path.join(directory, 'sitemap.xml'), 'w') as f:
-                f.write(xml)
+                f.write(smart_str(render_to_string('wiki/sitemap.xml',
+                                  {'urls': WikiSitemap(info).get_urls(page=1)})))
+
+            # Force the gc to cleanup
+            del queryset
```
Comment (robhudson) [inline kuma/wiki/tasks.py]: This only happens if count > 0. Should it be outside that loop?

File: kuma/wiki/tasks.py [hunk 3]
```diff
@@ -281,25 +285,29 @@ def build_sitemaps():
...
+            del info
 
             sitemap_url = absolutify('/sitemaps/%s/sitemap.xml' % locale)
-            sitemap_index = (sitemap_index + sitemap_element %
-                             (sitemap_url, timestamp))
+            sitemap_parts.append(SITEMAP_ELEMENT % (sitemap_url, timestamp))
 
-    sitemap_index = sitemap_index + "</sitemapindex>"
+    sitemap_parts.append(u"</sitemapindex>")
```
Comment (robhudson) [inline kuma/wiki/tasks.py]: everywhere else you've used single quotes

**Comment by robhudson** [comment]
r+wc.

File: kuma/wiki/tasks.py (hunk 3 above)
Comment (jezdez) [inline kuma/wiki/tasks.py]: god damnit

File: kuma/wiki/tasks.py (hunk 2 above)
Comment (jezdez) [inline kuma/wiki/tasks.py]: ah, yeah

File: kuma/wiki/tasks.py (hunk 1 above)
Comment (jezdez) [inline kuma/wiki/tasks.py]: good point

**Pull request merged**